src/main/java/software/amazon/documentdb/jdbc/sshtunnel/DocumentDbSshTunnelServer.java

```diff
@@ -0,0 +1,736 @@
+        return command;
+    }
+
+    /**
+     * Gets the command line parameters for invoking the Java executable in the JAVA_HOME.
+     *
+     * @param className the name of the class that contains the 'main()' method.
+     * @param arguments the arguments to the main method.
+     * @return a list of arguments to pass to {@link ProcessBuilder}.
+     * @throws SQLException when the path to the java executable cannot be resolved.
+     */
+    public static List<String> getJavaCommand(final String className, final String... arguments)
+            throws SQLException, URISyntaxException {
+        final String javaBinFilePath = getJavaBinFilePath();
+        final String combinedClassPath = getCombinedClassPath();
+
+        final List<String> command = new LinkedList<>();
+        command.add(javaBinFilePath);
+        command.add(CLASS_PATH_OPTION_NAME);
+        command.add(combinedClassPath);
+        command.add(className);
+        command.addAll(Arrays.asList(arguments));
+        return command;
+    }
+
+    private static String getJavaBinFilePath() throws SQLException {
+        // Check that the java command executable is available relative to the
+        // JAVA_HOME environment variable.
+        final String javaHome = getJavaHome();
+        final String javaBinFilePath = Paths.get(javaHome, BIN_FOLDER_NAME, JAVA_EXECUTABLE_NAME).toString();
+        final boolean isOsWindows = org.apache.commons.lang3.SystemUtils.IS_OS_WINDOWS;
+        final Path javaBinPath = Paths.get(javaBinFilePath + (isOsWindows ? ".exe" : ""));
+        if (!Files.exists(javaBinPath) || !Files.isExecutable(javaBinPath)) {
+            throw SqlError.createSQLException(
+                    LOGGER,
+                    SqlState.CONNECTION_EXCEPTION,
+                    SqlError.MISSING_JAVA_BIN,
+                    javaBinPath.toString());
+        }
+        return javaBinFilePath;
+    }
+
+    private static String getJavaHome() throws SQLException {
+        final String javaHome = System.getProperty(JAVA_HOME);
+        if (isNullOrWhitespace(javaHome) || !Files.exists(Paths.get(javaHome))) {
+            throw SqlError.createSQLException(
+                    LOGGER,
+                    SqlState.CONNECTION_EXCEPTION,
+                    SqlError.MISSING_JAVA_HOME);
+        }
+        return javaHome;
+    }
+
+    private static String getCombinedClassPath() throws URISyntaxException {
+        URI currentClassPathUri = DocumentDbSshTunnelServer.class
+                .getProtectionDomain().getCodeSource().getLocation().toURI();
+        final String schemeSpecificPart = currentClassPathUri.getSchemeSpecificPart();
+        if (currentClassPathUri.getScheme().equalsIgnoreCase(FILE_SCHEME)
+                && !isNullOrWhitespace(schemeSpecificPart)) {
+            // Ensure only 1 slash at beginning.
+            final String startsWithSlashExpression = "^/+";
+            currentClassPathUri = new URI(currentClassPathUri.getScheme()
```

Comment:
This handles the late-bound invocation and ensures the URI is in the correct syntax.